jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/ContainerHandlerTest.java

```diff
@@ -195,13 +195,13 @@ void withImageNames_shouldReturnContainersWithConfiguredImages() {
       }
 
       @Test
-      @DisplayName("with user, image and tag with period in image user, should return configured image")
-      void withUserAndImageAndTagWithPeriodInImageUser_shouldReturnConfiguredImage() {
+      @DisplayName("with user, image and tag with dash in image user, should return configured image")
+      void withUserAndImageAndTagWithDashInImageUser_shouldReturnConfiguredImage() {
```

Comment:
Is it necessary to modify existing tests? I remember this test was related to https://github.com/eclipse/jkube/issues/523 in which the user name contained `.` 

You can add additional test to cover your use case.

Reply:
I think it is not correct to have a . in the "username" part

Reply:
I think problem is more complicated.
Basically I see those cases:
- ecplise_jkube - there is one part, it's a docker hub registry
- eclipse/ecplise_jkube - there are 2 parts; it's docker hub registry; eclipse repo
- user.jfrog/eclipse_jkube - there are 2 parts; it's jfrog registry;
- quay.io/my.user/eclipse_jkube - there are 3 parts; it's quay.io registry; my.user username
I think my fix is correct.
Username can have dots if it's not a docker hub registry. Based on dot in 2 parts image name it's possible to determine if it's registry or username.
3 parts image name can have dots.
However there is a logic that adds registry after image name was parsed (eg ImageName.getFullName(String optionalRegistry)). We cannot add registry once image name was parsed as it's not possible to determine if we dealing with registry or username.
ImageName.getFullName is used in combination with EnvUtil.firstRegistryOf(imageName.getRegistry(), registryConfig.getRegistry());
How can it parse ImageName properly if registry is selected based on ImageName or RegistryConfig?

BTW there is another case
- user.jfrog/my-repo/eclipse_jkube - there are 3 parts; it's jforg registry; user is a username; my-repo is a repo
I would park this case as user can explicitly provide username

Reply:
> I think it is not correct to have a . in the "username" part

We did have an issue (#523) and needed to provide a fix for this specific case `roman.fake/customer-service-cache:latest` 

So we need to ensure that the fix provided for the scenario described in the issue is still applicable

Reply:
https://docs.docker.com/engine/reference/commandline/tag/

> An image name is made up of slash-separated name components, optionally prefixed by a registry hostname. The hostname must comply with standard DNS rules, but may not contain underscores. If a hostname is present, it may optionally be followed by a port number in the format :8080. If not present, the command uses Docker’s public registry located at registry-1.docker.io by default. Name components may contain lowercase letters, digits and separators. A separator is defined as a period, one or two underscores, or one or more dashes. A name component may not start or end with a separator.
> 
> A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits, underscores, periods and dashes. A tag name may not start with a period or a dash and may contain a maximum of 128 characters.

Reply:
Then there is no way to determine if it's a username or service domain name.
repo.jfrog.io/eclipse_jkube vs roman.fake/eclipse_jkube 
I see below solutions:
- add special case for jfrog parsing
- do some extra logic to do some fall back logic for download
- add section to the documentation and error message to explicitly add port number eg repo.jfrog.io:80/eclipse_jkube

Reply:
@manusa @sunix : thoughts??

Reply:
in Dockerhub from what I see (when creating an account), a username can have `4 to 30 letters & digits only`.

Reply:
In all the cases, I am wondering how clients like `docker` or `podman` behave in each cases. We should rely on it maybe.